posh-git.psm1

```diff
@@ -33,83 +33,71 @@ else {
     $defaultPromptDef = [Runspace]::DefaultRunspace.InitialSessionState.Commands['prompt'].Definition
 }
 
-# If there is no prompt function or the prompt function is the default, export the posh-git prompt function.
+# If there is no prompt function or the prompt function is the default, replace the current prompt function definition
 $promptReplaced = $false
-$origPsrlExtraPromptLineCount = 0
 $poshGitPromptScriptBlock = $null
 
 $currentPromptDef = if ($funcInfo = Get-Command prompt -ErrorAction SilentlyContinue) { $funcInfo.Definition }
 if (!$currentPromptDef -or ($currentPromptDef -eq $defaultPromptDef)) {
-    $poshGitPromptScriptBlock = {
-        $origLastExitCode = $LASTEXITCODE
+    # Have to use [scriptblock]::Create() to get debugger detection to work in PS v2
+    $poshGitPromptScriptBlock = [scriptblock]::Create(@'
+        $origLastExitCode = $global:LASTEXITCODE
 
         # A UNC path has no drive so it's better to use the ProviderPath e.g. "\\server\share".
         # However for any path with a drive defined, it's better to use the Path property.
         # In this case, ProviderPath is "\LocalMachine\My"" whereas Path is "Cert:\LocalMachine\My".
         # The latter is more desirable.
         $pathInfo = $ExecutionContext.SessionState.Path.CurrentLocation
-        $curPath = if ($pathInfo.Drive) { $pathInfo.Path } else { $pathInfo.ProviderPath }
-        if ($curPath -and $curPath.ToLower().StartsWith($Home.ToLower()))
+        $currentPath = if ($pathInfo.Drive) { $pathInfo.Path } else { $pathInfo.ProviderPath }
+
+        # File system paths are case-sensitive on Linux and case-insensitive on Windows and macOS
+        if (($PSVersionTable.PSVersion.Major -ge 6) -and $IsLinux) {

```

Comment:
Micro-optimization, but we could calculate `$stringComparison` once, assuming we can get the scriptblock to close over the variable.